### PR TITLE
gh-123836: check zero signs in math_testcases.txt

### PR DESCRIPTION
**Pull request by skirpichev**
Just like cmath_testcases.txt.  These tests require IEEE 754 anyway.

* Issue: gh-123836

### CODE REVIEW

**Comment by skirpichev** [comment]
Probably could skip news.

**Comment by skirpichev** [comment]
CC @picnixz

**Comment by picnixz** [comment]
Sorry Sergey for not replying to your pings. I see them but I'm either doing something else or then I forget about them :(

File: Lib/test/test_math.py [hunk 1]
```diff
@@ -2053,6 +2056,12 @@ def test_testfile(self):
                 if id in SKIP_ON_TIGER:
                     continue
 
+            # Skip some sqrt tests.  C99+ says for math.h's sqrt: If the
+            # argument is +∞ or ±0, it is returned, unmodified.  On another
+            # hand, for csqrt: If z is ±0+0i, the result is +0+0i.
+            if id in ['sqrt0002', 'sqrt0003', 'sqrt1001', 'sqrt1023']:
+                continue
```
Comment (vstinner) [inline Lib/test/test_math.py]: Rather than skipping the test, would it be possible to only skip the zero sign check?

Comment (skirpichev) [reply Lib/test/test_math.py]: Yes, I did this by fixing sign of er.

**Review by vstinner — APPROVED** [review]
LGTM

**Comment by vstinner** [comment]
Merged, thanks.

**Comment by vstinner** [comment]
Oh, a test failed on Windows 10:

```
FAIL: testFmod (test.test_math.MathTests.testFmod)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\buildarea\3.x.bolen-windows10\build\Lib\test\test_math.py", line 605, in testFmod
    self.ftest('fmod(-10, 1)', math.fmod(-10, 1), -0.0)
    ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\buildarea\3.x.bolen-windows10\build\Lib\test\test_math.py", line 258, in ftest
    self.fail("{}: {}".format(name, failure))
    ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: fmod(-10, 1): expected -0.0, got 0.0 (zero has wrong sign)
```

**Comment by skirpichev** [comment]
It's easy: windows=>bugs.  Standard says: "The fmod functions compute the floating-point remainder of x/y. [...] if y is nonzero, the result has the same sign as x [...]".

Does it fail on 11?  Maybe it's fixed?

Anyway, I'll provide patch to filter out this on Win 10.

**Comment by vstinner** [comment]
> Anyway, I'll provide patch to filter out this on Win 10.

Is it possible to fix fmod() bug on Windows?

**Comment by skirpichev** [comment]
Unfortunately, I can only guess how it's broken.

Assuming that it's wrong only for signed zero as result, we could replace one with
```c
static double
m_fmod(double x, double y)
{
    double r = fmod(x, y);

    if (r == 0.0 && y != 0.0) {
        r = copysign(r, x);
    }

    return r;
}
```

**Comment by vstinner** [comment]
> Does it fail on 11? Maybe it's fixed?

The test pass on [MSC v.1941 64 bit (AMD64)] (Windows 10 buildbot), but fails on [MSC v.1916 64 bit (AMD64)] (GitHub Actions). It's more a matter of the C Compiler version (MSC).

**Pull request merged**